swag-2024-Q1-openday/slides.md

```diff
@@ -0,0 +1,296 @@
+# Packet Delivery Time
+
+$$PktTsbpdTime = TsbpdTimeBase + PktTimestamp + TsbpdDelay + Drift$$
+
+- $$TsbpdTimeBase$$ : Time base reflects the time difference between local clock of the receiver and the sender
+- $$PktTimestamp$$ : Data packet timestamp
+- $$TsbpdDelay$$ : SRT Latency, negotiated between sender and receiver on handshake.
+- $$Drift$$ : Adjust the fluctuations between sender and receiver clock.
+
+---
+
+| Serial Number | PktTimestamp | Time Base    | RCV Clock    | SRT Latency | Drift | Packet Delivery Time |
+|---------------|--------------|--------------|--------------|-------------|-------|----------------------|
+| 1             | 20           | 00:00:00,040 | 00:00:00,060 | 120         | 0     | 00:00:00,180         |
+| 2             | 40           | 00:00:00,040 | 00:00:00,080 | 120         | 0     | 00:00:00,200         |
+| 5             | 100          | 00:00:00,040 | 00:00:00,140 | 120         | 0     | 00:00:00,260         |
+| 3             | 60           | 00:00:00,040 | 00:00:00,210 | 120         | 0     | 00:00:00,220         |
+| 4             | 80           | 00:00:00,040 | 00:00:00,212 | 120         | 0     | 00:00:00,240         |
+
+^ 想像Receiver 端收到1~5 packate, 計算Packet Delivery Time 的範例。
+
+---
+[.build-lists: true]
+# [fit] Quest 2: packet delivery time 
+
+| Serial Number | PktTimestamp | Time Base    | RCV Clock    | SRT Latency | Drift | Packet Delivery Time |
+|---------------|--------------|--------------|--------------|-------------|-------|----------------------|
+| 6             | 120          | 00:00:00,040 | 00:00:00,160 | 120         | 0     | 00:00:00,???         |
+| 7             | 140          | 00:00:00,040 | 00:00:00,180 | 120         | 0     | 00:00:00,???         |
+
+
+- 6: 00:00:00,280
+- 7: 00:00:00,300
+
+
+---
+[.footer: https://qiita.com/tomoyafujita/items/2e10a9b9d463a36d4a3e]
+
+![inline](srt-too-late-packet-drop.png)
+
+
+---
+[.build-lists: true]
+![original](background.png)
+# Too Late Packet
+- Sender will drop packets that have no chance to be delivered in time.
+- Packet is considered "too late" if the packet timestamp is older than **TLPKTDROP_THRESHOLD**
+- Receiver will "skip" this packet and send a fake ACK packet to the sender
+
+---
+# [fit] Quest 3: Starts from with is too late ?
+[.build-lists: true]
+
+| Serial Number    | PktTimestamp | Time Base    | RCV Clock    | SRT Latency | Drift | TLPKTDROP_THRESHOLD |
+|------------------|--------------|--------------|--------------|-------------|-------|---------------------|
+| 3                | 60           | 00:00:00,040 | 00:00:00,210 | 120         | 0     | 150                 |
+| 3 (retransmit-1) | 60           | 00:00:00,040 | 00:00:00,235 | 120         | 0     | 150                 |
+| 3 (retransmit-2) | 60           | 00:00:00,040 | 00:00:00,260 | 120         | 0     | 150                 |
+| 3 (retransmit-3) | 60           | 00:00:00,040 | 00:00:00,285 | 120         | 0     | 150                 |
```

Comment:
- Need example
- Move "SRT Latency" and "TLPKTDROP_THRESHOLD" to be out of the table
- Move RCV Clock to front